codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/UserAgentMiddleware.kt

```diff
@@ -39,11 +39,11 @@ class UserAgentMiddleware(val settings: SwiftSettings) : MiddlewareRenderable {
         )
     }
 
-    private fun middlewareParamsString(writer: SwiftWriter): String {
+    private fun middlewareParamsString(ctx: ProtocolGenerator.GenerationContext, writer: SwiftWriter): String {
         return writer.format(
-            "metadata: \$N.fromConfig(serviceID: serviceName, version: \$S, config: config)",
+            "metadata: \$N.fromConfig(serviceID: serviceName, version: \$L.version, config: config)",
             AWSClientRuntimeTypes.Core.AWSUserAgentMetadata,
-            settings.moduleVersion,
+            ctx.symbolProvider.toSymbol(ctx.service).name,
```

Comment:
Instead of writing a literal string with the user-agent version to the middleware initializer, use the service's version property to help DRY up the code.

Reply:
See test expectations below for new generated code.